src/doc/unstable-book/src/compiler-flags/annotate-moves.md

```diff
@@ -0,0 +1,79 @@
+# `annotate-moves`
+
+The `-Z annotate-moves` flag enables annotation of compiler-generated
+move and copy operations, making them visible in profilers and stack traces
+for performance debugging.
+
+When enabled, the compiler manipulates debug info to make large move and copy
+operations appear as if they were inlined calls to `core::profiling::compiler_move`
+and `core::profiling::compiler_copy`. No actual function calls are generated -
+this is purely a debug info transformation that makes expensive memory operations
+visible in profilers and stack traces.
+
+## Syntax
+
+```bash
+rustc -Z annotate-moves[=<value>]
+```
+
+Where `<value>` can be:
+- A boolean: `true`, `false`, `yes`, `no`, `on`, `off`
+- A number: size threshold in bytes (e.g., `128`)
+- Omitted: enables with default threshold (65 bytes)
+
+## Options
+
+- `-Z annotate-moves` or `-Z annotate-moves=true`: Enable with default size limit
+- `-Z annotate-moves=false`: Disable annotation
+- `-Z annotate-moves=N`: Enable with custom size limit of N bytes
+
+## Examples
+
+```bash
+# Enable annotation with default threshold (65 bytes)
+rustc -Z annotate-moves main.rs
+
+# Enable with custom 128-byte threshold
+rustc -Z annotate-moves=128 main.rs
+
+# Only annotate very large moves (1KB+)
+rustc -Z annotate-moves=1024 main.rs
+
+# Explicitly disable
+rustc -Z annotate-moves=false main.rs
+```
+
+## Behavior
+
+The annotation only applies to:
+- Types larger than the specified size threshold
+- Non-immediate types (those that would generate `memcpy`)
+- Operations that actually move/copy data (not ZST types)
```

Comment:
What happens if you specify a threshold like 2 then? I assume that there wouldn't be any profiling functions called with that size. Or would you still see copies/moves to/from memory? If not, maybe we should issue a warning if a threshold < 64 is specified?

Reply:
It would also help if you had a test that tested for these kinds of things - e.g. tested that SIMD types aren't annotated, or tested that a `u32` move isn't annotated even if the threshold is very low.